Add docs for ListTop builtins (#4795)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/list.md

{% endif %}

## ListTop, ListTopAsc, ListTopDesc, ListTopSort, ListTopSortAsc и ListTopSortDesc {#listtop}

Select top values from the list. `ListTopSort*` additionally sorts the returned values. The smallest values are selected by default. Thus, the functions without a suffix are the aliases to `*Asc` functions, while `*Desc` functions return the largest values.

> ListTopSort is more effective that consecutive ListTop and ListSort (because ListTop can partially sort the list in order to find needed values), however ListTop is more effective than ListTopSort when the order of the result is not important.

Reviewer: ```suggestion
`ListTopSort` is more effective than consecutive `ListTop` and `ListSort` because `ListTop` can partially sort the list to find needed values. However, `ListTop` is more effective than `ListTopSort` when the result order is unimportant.
```

{% endif %}

## ListTop, ListTopAsc, ListTopDesc, ListTopSort, ListTopSortAsc и ListTopSortDesc {#listtop}

> Select top values from the list (ListTopSort* - additionally sort the list). The smallest values are selected by default (the functions without a suffix are the aliases to *Asc functions; *Desc - selection of the biggest values).

Reviewer: The difference between variants with and without the Sort suffix is unclear: I could assume that without it, a function finds the Nth element (like in [std::nth_element](https://en.cppreference.com/w/cpp/algorithm/nth_element)) and returns everything less than it, without an actual complete sort. Regardless of whether this guess is correct or not, let's explicitly explain how it works and what benefit to expect by omitting the Sort suffix.

Author: Немного поправил эту неточность

> Select top values from the list (ListTopSort* - additionally sort the list). The smallest values are selected by default (the functions without a suffix are the aliases to *Asc functions; *Desc - selection of the biggest values).

Reviewer: ```suggestion
Select top values from the list. `ListTopSort*` additionally sorts the list. The smallest values are selected by default. Thus, the functions without a suffix are the aliases to `*Asc` functions, while `*Desc` functions return the largest values.
```

Arguments:

1. List.
2. Size of selection.
3. An optional expression to get the sort key from a list element (it's the element itself by default).

**Signature**
```
ListTop(List<T>, N)->List<T>

> ListTop(List<T>?, N)->List<T>?

Reviewer: This behavior is called `{Flags:AutoMap}` in neighboring articles like https://ydb.tech/docs/en/yql/reference/udf/list/datetime

Author: Кажется, это не совсем AutoMap, хоть и похоже. AutoMap применим скорее только к Udf, нежели ко встроенным функциям. А структура описания этой функции повторяет структуру описания ListSort

Reviewer 2: да, я поддерживаю описать сигнатуру в виде
ListTop(AutoMap<List<T>>, N)->List<T>
